### DISCUSSION TITLE
Cluster Provisioning using Flux with Cluster API

### DISCUSSION THREAD

**Discussion by vijayansarathy** [fluxcd/flux2#1832] [General] [answered]
Hi,
I see the following at https://fluxcd.io/

_Multi-cluster infrastructure and apps work out of the box with Cluster API: Flux can use one Kubernetes cluster to manage apps in either the same or other clusters, spin up additional clusters themselves, and manage clusters including lifecycle and fleets._

I have used Flux (v2) quite a bit for deploying/managing application and infrastructure workloads across multiple clusters. But I have not used Flux for cluster provisioning per se. I don't see anything in the documentation about how to do this. Can someone please point me in the right direction? 
Looking for working samples of using Cluster API Provider for AWS to provision either a Kubernetes or EKS cluster.

Thanks,
Viji

**Comment by stefanprodan** [comment, marked as answer]
You can run Flux on your management cluster, push cluster definitions to git, then Flux will create the clusters, then it will connect to them using the CAPI generated kubeconfig and reconcile workloads on those remote clusters, see:
- https://fluxcd.io/docs/components/kustomize/kustomization/#remote-clusters--cluster-api
- https://fluxcd.io/docs/components/helm/helmreleases/#remote-clusters--cluster-api
